mlflow/store/sftp_artifact_repo.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import TempDir
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+
+class SFTPArtifactRepository(ArtifactRepository):
+    """Stores artifacts as files in a remote directory, via sftp."""
+
+    def __init__(self, artifact_uri, client=None):
+        self.uri = artifact_uri
+        parsed = urlparse(artifact_uri)
+        self.config = {
+            'host': parsed.hostname,
+            'port': 22 if parsed.port is None else parsed.port,
+            'username': parsed.username,
+            'password': parsed.password
+        }
+        self.path = parsed.path
+
+        if client:
+            self.sftp = client
+        else:
+            import pysftp
+            import paramiko
+
+            ssh_config = paramiko.SSHConfig()
+            user_config_file = os.path.expanduser("~/.ssh/config")
+            if os.path.exists(user_config_file):
+                with open(user_config_file) as f:
+                    ssh_config.parse(f)
+
+            user_config = ssh_config.lookup(self.config['host'])
```

Comment:
Does this work if `self.config['host']` is `None`, e.g. if someone uses a URI like `sftp:///some/path`? Looks like `lookup` might expect a string (see [Paramiko API docs](http://docs.paramiko.org/en/2.4/api/config.html#paramiko.config.SSHConfig.lookup))

Reply:
This was indeed an issue. I've added a default case. If `self.config['host'] is None`, `self.config['host'] = 'localhost'`.